src/vs/vscode.proposed.d.ts

```diff
@@ -1524,15 +1525,20 @@ declare module 'vscode' {
 		data: { [key: string]: any };
 	}
 
-	export type CellOutput = CellStreamOutput | CellErrorOutput | CellDisplayOutput;
+	export type CellOutput = CellStreamOutput | CellErrorOutput | CellDisplayOutput & { metadata: { [key: string]: any} }; // Metadata can be put here by an execution so that a renderer can use it
+
+	export interface NotebookExecutionInfo {
+		data: { [key: string]: any}
+	}
 
 	export interface NotebookCell {
 		readonly uri: Uri;
 		handle: number;
 		language: string;
 		cellKind: CellKind;
 		outputs: CellOutput[];
-		getContent(): string;
+		executionInfo: NotebookExecutionInfo;
```

Comment:
How about we call this `data`.
Something that gets passed around, akin to `metadata`.
How it is used is upto the extension authors.

Reply:
Good idea